packages/next/src/build/webpack-config.ts

```diff
@@ -1514,8 +1514,7 @@ export default async function getBaseWebpackConfig(
         return `module ${request}`
       }
 
-      // Other Next.js internals need to be transpiled.
-      return
+      return resolveNextExternal(request)
```

Comment:
Why it's going through `resolveNextExternal` again instead of marking as "should be transpiled" with returning `undefined`

Reply:
because in some cases `resolveNextExternal` resolves to an external, like in the test scenario. In the other scenarios it returns `undefined` which is fine

Reply:
Was thinking the same but `request` is a different parameter than the one passed to `resolveNextExternal` initially.

Reply:
sounds good, would be great to add a comment to explain the recursion case here

Reply:
it looks confusing from the UI but the change at line 1517 is not part of the `resolveNextExternal` function, it's part of `handleExternals`.